### PR TITLE
Fix mongodb database write prohibition plugin bug

### PR DESCRIPTION
**Pull request by daizhenyu**
【Issue】#1448

【Content】Fix mongodb database write prohibition plugin bug during function tests

【Case Description】No need

【Self-Test Result】1、The local static check is passed.

【Scope of Influence】Follow-up supplementary documents

### CODE REVIEW

File: report/pom.xml [hunk 1]
```diff
@@ -48,9 +48,24 @@
...
+        </dependency>
+        <dependency>
+            <groupId>com.huaweicloud.sermant</groupId>
+            <artifactId>mysql-mariadb-2.x-plugin</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.huaweicloud.sermant</groupId>
+            <artifactId>mysql-mariadb-3.x-plugin</artifactId>
+            <version>1.0.0</version>
```
Comment (hanbingleixue) [inline report/pom.xml]: 使用${project.version}

File: sermant-plugins/sermant-database-write-prohibition/mongodb-4.x-plugin/src/main/java/com/huaweicloud/sermant/mongodbv4/utils/MongoDbV4EnhancementHelper.java [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+    }
+
+    /**
+     * Get Parametric Interceptor of SyncOperationHelper executeRetryableWrite Method
+     *
+     * @param handler write operation handler
+     * @return InterceptDeclarer Parametric Interceptor of SyncOperationHelper executeRetryableWrite Method
+     */
+    public static InterceptDeclarer getExecuteRetryableWriteInterceptDeclarer4Sync(DatabaseHandler handler) {
+        return InterceptDeclarer.build(getExecuteCommandMethodMatcher(),
```
Comment (hanbingleixue) [inline sermant-plugins/sermant-database-write-prohibition/mongodb-4.x-plugin/src/main/java/com/huaweicloud/sermant/mongodbv4/utils/MongoDbV4EnhancementHelper.java]: 拦截executeRetryableWrite 方法应该使用getExecuteRetryableWriteMethodMatcher

File: report/pom.xml (hunk 1 above)
Comment (daizhenyu) [reply report/pom.xml]: 已修改

File: sermant-plugins/sermant-database-write-prohibition/mongodb-4.x-plugin/src/main/java/com/huaweicloud/sermant/mongodbv4/utils/MongoDbV4EnhancementHelper.java (hunk 2 above)
Comment (daizhenyu) [reply sermant-plugins/sermant-database-write-prohibition/mongodb-4.x-plugin/src/main/java/com/huaweicloud/sermant/mongodbv4/utils/MongoDbV4EnhancementHelper.java]: 已修改

File: sermant-plugins/sermant-database-write-prohibition/mongodb-3.x-plugin/src/main/java/com/huaweicloud/sermant/mongodbv3/interceptors/GeneralExecuteInterceptor.java [hunk 3]
```diff
@@ -14,52 +14,49 @@
...
-        ConnectionDescription description = connection.getDescription();
-        if (description != null) {
-            ServerAddress serverAddress = description.getServerAddress();
-            info.setHostAddress(serverAddress.getHost());
-            info.setPort(serverAddress.getPort());
-        }
+        ServerAddress serverAddress = ((WriteBinding) context.getArguments()[0])
+                .getWriteConnectionSource()
+                .getServerDescription()
+                .getAddress();
```
Comment (lilai23) [inline sermant-plugins/sermant-database-write-prohibition/mongodb-3.x-plugin/src/main/java/com/huaweicloud/sermant/mongodbv3/interceptors/GeneralExecuteInterceptor.java]: 多层get可能有空指针，考虑
serverAddress = Optional.ofNullable(writeBinding).map(WriteBinding::getWriteConnectionSource).map(WriteBinding::getServerDescription).map(WriteBinding::getAddress).orElse(xxx);

File: sermant-plugins/sermant-database-write-prohibition/mongodb-3.x-plugin/src/main/java/com/huaweicloud/sermant/mongodbv3/interceptors/MixedBulkWriteOperationInterceptor.java [hunk 4]
```diff
@@ -14,52 +14,50 @@
...
-        ConnectionDescription description = connection.getDescription();
-        if (description != null) {
-            ServerAddress serverAddress = description.getServerAddress();
-            info.setHostAddress(serverAddress.getHost());
-            info.setPort(serverAddress.getPort());
-        }
+        ServerAddress serverAddress = ((WriteBinding) context.getArguments()[0])
+                .getWriteConnectionSource()
+                .getServerDescription()
+                .getAddress();
```
Comment (lilai23) [inline sermant-plugins/sermant-database-write-prohibition/mongodb-3.x-plugin/src/main/java/com/huaweicloud/sermant/mongodbv3/interceptors/MixedBulkWriteOperationInterceptor.java]: 同上

File: sermant-plugins/sermant-database-write-prohibition/mongodb-3.x-plugin/src/main/java/com/huaweicloud/sermant/mongodbv3/interceptors/GeneralExecuteInterceptor.java (hunk 3 above)
Comment (daizhenyu) [reply sermant-plugins/sermant-database-write-prohibition/mongodb-3.x-plugin/src/main/java/com/huaweicloud/sermant/mongodbv3/interceptors/GeneralExecuteInterceptor.java]: 已修改

File: sermant-plugins/sermant-database-write-prohibition/mongodb-3.x-plugin/src/main/java/com/huaweicloud/sermant/mongodbv3/interceptors/MixedBulkWriteOperationInterceptor.java (hunk 4 above)
Comment (daizhenyu) [reply sermant-plugins/sermant-database-write-prohibition/mongodb-3.x-plugin/src/main/java/com/huaweicloud/sermant/mongodbv3/interceptors/MixedBulkWriteOperationInterceptor.java]: 已修改

**Review by lilai23 — APPROVED** [review]

**Review by hanbingleixue — APPROVED** [review]

**Pull request merged**